### ISSUE TITLE
Options need to be registered twice to use

### ISSUE THREAD

**Issue by 0x1af2aec8f957** [timreichen/cli#4]
#### I don't expect options to register twice。

Example:

> file: import_map.json
```json
"imports": {
    "program": "https://deno.land/x/program/mod.ts"
  }
```
> file: cli.ts
```ts
import * as commandModule from './commands/index.ts';
import * as optionModule from './options/index.ts';
// import { /* fileDeps, */ commandInsert } from './utils/common.ts';

const program = new Program({
    name: "ts-cli",
    description: "ts-cli.",
    version: "0.0.1"
});

// commandInsert func:
/* function commandInsert(program: Program, option: ConfigType.CommandExtend): void{
    const {options, args, commands}: ConfigType.CommandExtend = option;

    if (options) {
        for(const [__name, option] of Object.entries(options)){
            program.option(<ConfigType.Option>{
                name: __name,
                ...<object>option
            });
        }
    }

    if (args){
        for(const arg of args){
            program.argument(<ConfigType.Argument>arg);
        }
    }

    if (commands){
        for(const [__name, command] of Object.entries(commands)){
            const _command = <{[name: string]: ConfigType.Command}><unknown>command;

            const _program: Program = program.command(<ConfigType.Command>{
                name: __name,
                ..._command
            });

            if (_command.commands) commandInsert(_program, _command.commands);
        }
    }
} */

for (const [name, option] of Object.entries(optionModule)/*fileDeps(pathJoin(workDir, 'options'), COMMAND_EXTNAME)*/){

    program.option(<ConfigType.Option>{
        name,
        ...option.default
    });
}

for (const [name, command] of Object.entries(commandModule) /*fileDeps(pathJoin(workDir, 'commands'), COMMAND_EXTNAME)*/){

    const {
        commands,
        options,
        args,
        ...commandOption
    } = <ConfigType.CommandExtend>command.default;

    commandInsert(program.command(<ConfigType.Command>{
        name,
        ...commandOption
    }), {
        commands,
        options,
        args,
    })
}

program.parse(Deno.args);
```
> file: ./commands/index.ts
```ts
// export * as commit from './commit.ts';
export * as create from './create.ts';
// export * as doctor from './doctor.ts';
// export * as install from './install.ts';
```

> file: ./commands/create.ts
```ts
import { ConfigType } from '../plugins/typings.d.ts';
import Git from '../plugins/git-cli.ts';
import { VUE_TEMPLATE_ADDRESS } from '../utils/constant.ts';

export default {
    // name: "create",
    description: "create-test",
    commands:/* <ConfigType.Command> */{},
    options:/* <{[name: string]: ConfigType.Option}> */{
        type : {
           alias: "t",
           // boolean: true,
           description: "name-type-option"
       }
    },
    args: <ConfigType.Argument[]>[{
        name: 'name',
        description: 'name-test'
    }],
    async fn(params: any){
        // if (args.quiet) return;
        const {_: args, ...options} = params;
        const [name] = <string []>args;

        const git: Git = Git.generateGitInstance(VUE_TEMPLATE_ADDRESS);
        const {status, message} = await git.clone(name, true);
        if (!status){
            throw new Error(message);
        }
    }
}
```

> file: ./options/index.ts
```ts
// export * as quiet from './quiet.ts';
export * as type from './type.ts';
```

> file: ./options/type.ts
```ts
export default {
    // name: "type",
    alias: "t",
    description: "type-test",
    boolean: true,
}
```

#### If I do not register the type in program.option, just register in the command, it will report an error:

```bash
deno run --unstable --reload --allow-all --importmap=import_map.json cli.ts create -t=vue vue-test
```

```bash
error: Found argument '--t' which wasn't expected, or isn't valid in this context
```

**Comment by timreichen** [comment]
Options can be registered either to program (which is the root command) or to subcommands.
If you want to use ```-t, --type``` on the subcommand you need to call option on
```ts
program.command(…)
.option(…)
```
In your code the options are registered to program here
```ts
 program.option(<ConfigType.Option>{
  name,
  ...option.default
});
```
<s>If you want to register  ```-t, --type``` on program and also call the subcommand, you need to put the option in front of the subcommand like so:
```sh
deno run --unstable --reload --allow-all --importmap=import_map.json cli.ts -t=vue create vue-test

```
</s>

**Comment by 0x1af2aec8f957** [comment]
@timreichen 

#### cli.ts content:
```ts
import { Program } from 'program';

const program = new Program({
    name: "ts-cli",
    description: "ts-cli.",
    version: "0.0.1"
});

function log(args: any) {
    // if (args.quiet) { return }
    console.log(args)
}

program
    .command({ name: "log", description: "logs parsed arguments", fn: log })
    .option({ name: "quiet", alias: "q", description: "Suppress diagnostic output" })
    .argument({ name: "source_file", multiple: true, optional: true })



program.parse(Deno.args);
```


#### import_map.json content:

```json
{
  "imports": {
    "program": "https://deno.land/x/program/mod.ts"
  }
}
```

#### When I use the option parameter, an error occurs:

```bash
deno run --unstable --reload --allow-all --importmap=import_map.json cli.ts -q=test log
```

```bash
error: Found argument '--q' which wasn't expected, or isn't valid in this context
```

#### But when I use program.option alone to register quiet, these errors will be eliminated.

**Comment by timreichen** [comment]
Validation order is fixed in 0.1.4.

Sorry I commented something wrong above. You cannot pass an option to program while calling a subcommand.
`cli.ts -q=test log` is therefore invalid.

**Comment by 0x1af2aec8f957** [comment]
@timreichen nice!!!
